components/omega/doc/design/VerticalMixingCoeff.md

```diff
@@ -0,0 +1,182 @@
+
+## 2 Requirements
+
+### 2.1 Requirement: Vertical mixing coefficient interface should be modular
+
+To prepare for additional mixing closures and later improvements, the interface to vertical mixing must allow for easy connection of closures. It is assumed that the strength of vertical mixing from each closure is additive, such that the final vertical diffusion coefficient is the sum of the coefficients and potential non local terms from each closure.
+
+### 2.2 Requirement: Vertical mixing models should be have local and/or gradient free terms
+
+For optimal performance, initial models of vertical turbulent fluxes for Omega should be cast as an implicit, down gradient mixing and an explicit, gradient free, component, e.g.,
+
+$$
+\overline{w' \phi'}\approx \kappa(\overline{\rho},\overline{u},\overline{v}) \left(\frac{\partial \overline{\phi}}{\partial z} + \gamma(\overline{\rho},\overline{u},\overline{v}) \right)
+$$
+
+Here, $\phi$ is a generic tracer, $\overline{w'\phi'}$ is the vertical turbulent flux of that tracer, and $\gamma$ is the gradient free portion of the flux (often referred to as 'non-local'). The vertical diffusivity can be as simple as a constant value, to complex functions of quantities like shear and stratification. A similar equation can be written for turbulent momentum fluxes.
+
+The use of a flux term proportional to the local gradient allows this problem to be cast implicitly, allowing for larger time steps, reducing the cost of the mixing model.  When more complex schemes are considered, other techniques such as subcycling can be considered to improve performance.
+
+### 2.3 Requirement: Vertical mixing models cannot ingest single columns at a time
+
+Many standard vertical mixing libraries (e.g. CVMix) receive and operate on one column at a time. This results in poor performance, especially on accelerators. Any utilized model of vertical mixing must ingest and operate on multiple columns concurrently.
+
+## 3 Algorithmic Formulation
+
+For Omega-1 a few simple vertical mixing algorithms will be used.  
+
+### 3.1 Richardson number dependent mixing
+
+One of the simplest class of models of vertical turbulence are polynomial functions of the gradient Richardson number. As in MPAS-Ocean, we choose to define the Richardson number as
+
+$$
+Ri = \frac{N^2}{\left|\frac{\partial \mathbf{U}}{\partial z}\right|^2}
+$$
+
+where $N^2$ is the Brunt Vaisala Frequency, which for non Boussinesq flows is defined as
+
+$$
+N^2 = \frac{g}{\rho_0}\frac{\partial \rho}{\partial z}
+$$
+
+This term is discretized as
+
+$$
+N^2(k) = g \rho_0 \frac{\rho_{DD}(k) - \rho(k)}{z_m(k-1) - z_m(k)}
```

Comment:
Is `N^2` located at `k_bot` because we are using the gradient between `k` and `k-1`?

Reply:
$N^2$ is located/defined at the top of the cell. If we are cycling `k` from `minLevelCell + 1` to `maxLevelCell`, then a gradient between `k` and `k+1` would be the top of the cell. This is now stated.